src/aleph/http/core.clj

```diff
@@ -132,15 +139,17 @@
       default-value
       (if-let [e (find added k)]
         (val e)
-        (let [k' (str/lower-case (name k))
+        (let [k' (str/lower-case (if (keyword? k) (name k) k))
```

Comment:
Whenever you see crap like this, the reason is AsciiStrings.

Netty uses, and returns, AsciiStrings everywhere, because most protocols only use ASCII-range characters, so they save space and simplify certain algorithms (e.g., counting string length, changing case). AsciiStrings extend CharSequence, but not String, and some Clojure fns only work with String... like `name`. 

Try `(name (AsciiString. "toto"))` and you'll get a ClassCastException.

BTW, debugging this stuff in the early days was so annoying, I added a defmethod for `print-method` for AsciiString, so whenever an AsciiString is `pr`-ed out, it'll be prefixed with "AS", like:

```
(prn (AsciiString. "toto"))
AS"toto"
```

Otherwise they'd appear just like regular Strings at the REPL, which is mega-annoying.

The one consolation is AsciiStrings were designed to produce the same hash as regular Strings, or they'd wreak havoc when adding them to maps.